src/reference-manual/pedantic-mode.Rmd

```diff
@@ -0,0 +1,363 @@
+
+
+# Pedantic Mode
+
+Pedantic Mode is a compilation option built into Stanc3 that warns you about potential issues in your Stan program.
+
+For example, if you compile the following program with Pedantic Mode:
+
+    data {
+      int N;
+      real x[N];
+    }
+    parameters {
+      real sigma;
+    }
+    model {
+      real mu;
+      x ~ normal(mu, sigma);
+    }
+
+The compiler will print the following to stderr:
+
```

Comment:
There should not be references to `stderr` as that's not quite how Stan works.  The services make callbacks to functions passed in by the interfaces, which can then do whatever they like with it.  We send errors to one callback and non-errors to another.  I think CmdStan sends errors to stderr, but I'm not even sure about that.   R and Python intercept stderr and stdout to deal with for their REPL environments.